File: go/LeetCode/findMinArrowShots.go
package main

import "sort"

func main() {
	points := [][]int{{-2147483646,-2147483645},{2147483646,2147483647}}
	print(findMinArrowShots(points))

}

func findMinArrowShots(points [][]int) int {
	sort.Slice(points, func(i, j int) bool {
		if points[i][1]<points[j][1] {
			return true
		}else {
			return false
		}
	})
	res :=1
	end:=points[0][1]
	for i:=1;i<len(points);i++ {
		if points[i][0] > end {
			res++
			end = points[i][1]
		}
	}

	return res
}

File: cdd/LeetCode/LeetCode/findMinArrowShots.cpp
#include <vector>
#include <stdio.h>
#include <algorithm>

using namespace std;

class Solution {
public:
	int findMinArrowShots(vector<vector<int>>& points)
	{
		sort(points.begin(), points.end(), [](const auto& u, const auto& v) {
			return u[1]<v[1];
			});
		int res = 1;
		int end = points[0][1];
		for (int i = 1; i < points.size(); i++)
		{
			if (points[i][0]>end)
			{
				res++;
				end = points[i][1];
			}
		}
		return res;
	}
};